test/mitmproxy/addons/test_maplocal.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+
+from mitmproxy.addons.maplocal import MapLocal, file_candidates
+from mitmproxy.test import taddons
+from mitmproxy.test import tflow
+
+from mitmproxy.addons.modifyheaders import parse_modify_spec
+
+
+@pytest.mark.parametrize(
+    "url,spec,expected_candidates", [
+        (
+            "https://example.org/img/topic/subtopic/test.jpg",
+            ":example.com/foo:/tmp",
+            ["/tmp/img/topic/subtopic/test.jpg", "/tmp/img/topic/test.jpg", "/tmp/img/test.jpg", "/tmp/test.jpg"]
```

Comment:
I'm not sure if I understand this pattern. Why are we supposed to match here if `/foo` is in the subject?

Reply:
You're right, this wouldn't match in the `request` method, but `file_candidates` does not consider `subject`, so it's technically a valid test. It is confusing though, I'll change it (also see next comment).